libs/geometry/include/geometry/TangentSpaceMesh.h

```diff
@@ -284,6 +324,66 @@ class TangentSpaceMesh {
      */
     void getQuats(filament::math::quath* out, size_t stride = 0) const noexcept;
 
+    /**
+     * Get output auxiliary attributes.
+     * Assumes the `out` param is at least of getVertexCount() length (while accounting for
+     * `stride`).
+     *
+     * @param  out    Client-allocated array that will be used for copying out attribute as two
+     * floats.
+     * @param  stride Stride for iterating through `out`
+     */
+    void getAux(AuxAttribute attribute, filament::math::float2* out,
```

Comment:
Could we make `getAux` a template method and pre-instantiate all the specializations?

Reply:
Not without exposing implementation details unfortunately.  Or can you think of a way around that?

Reply:
You can do something like this:

```cpp
// In TangentSpaceMesh.h:
template<typename T>
void getAux(AuxAttribute attribute, T* out, size_t stride = 0) const noexcept;
```

```cpp
// In TangentSpaceMesh.cpp:
template void TangentSpaceMesh::getAux<filament::math::float2>(AuxAttribute attribute, filament::math::float2* out, size_t stride) const noexcept;
template void TangentSpaceMesh::getAux<filament::math::float3>(AuxAttribute attribute, filament::math::float3* out, size_t stride) const noexcept;
// etc

template<typename T>
void TangentSpaceMesh::getAux(AuxAttribute attribute, T* out, size_t stride) const noexcept {
    // ...
}
```

We do something similar inside [MaterialInstance.cpp](https://github.com/google/filament/blob/fc07e79c0d0882300634d3143488f4caa157ae5e/filament/src/MaterialInstance.cpp#L109-L123) for `setParameter`.

Reply:
nice! done